kore/app/exec/Main.hs

```diff
@@ -404,12 +407,13 @@ main = do
 mainWithOptions :: KoreExecOptions -> IO ()
 mainWithOptions execOptions = do
     let KoreExecOptions { koreLogOptions } = execOptions
-    exitCode <- catch (runLoggerT koreLogOptions go) $
+    exitCode <- runLoggerT koreLogOptions $ catch go $
         \(Goal.WithConfiguration lastConfiguration someException) -> do
-            renderResult
+            liftIO $ renderResult
                 execOptions
                 ("// Last configuration:\n" <> unparse lastConfiguration)
-            throwM someException
+            criticalExecutionError someException
```

Comment:
This actually needs to be lifted out to a higher handler because we might have an exception before there is any configuration. The body of the `mainWithOptions` should look something like,

```.hs
    exitCode <- runLoggerT koreLogOptions
        $ handle handleSomeException
        $ handle handleWithConfiguration
        $ go
```